src/sdam/description/topology/mod.rs

```diff
@@ -731,15 +741,17 @@ pub(crate) struct TopologyDescriptionDiff {
     pub(crate) new_addresses: HashSet<StreamAddress>,
 }
 
-fn verify_max_staleness(max_staleness: Option<Duration>) -> Result<()> {
+fn verify_max_staleness(max_staleness: Option<Duration>) -> crate::error::Result<()> {
```

Comment:
There were two existing uses of this function before this commit, one of which does return the error to users, so we still want to be able to return an actual Error rather than a string in this case. I split this function into two so that we don't have to pattern match the error and assert that it's an ArgumentError (when we know it always will be) when it's called in a place where we don't return the error to users.